src/dune_rules/compilation_context.mli

```diff
@@ -85,6 +86,8 @@ val package : t -> Package.t option
 
 val vimpl : t -> Vimpl.t option
 
+val lib_name : t -> Lib_name.t option
```

Comment:
How about changing the name to reflect that this is only for public libraries? `public_lib_name` for example.

Reply:
Good idea, changed it in bf43e9b